lib/whitehall_importer/create_revision.rb

```diff
@@ -111,15 +113,23 @@ def tags(associations)
       associations.map { |association| association["content_id"] }
     end
 
-    def create_image_revisions(images)
+    def find_or_create_image_revisions(images)
       images.reduce([]) do |memo, image|
-        memo << WhitehallImporter::CreateImageRevision.call(image, memo.map(&:filename))
+        memo << if (already_imported = WhitehallMigration::AssetImport.find_by(original_asset_url: image["url"]))
+                  already_imported.image_revision
+                else
+                  WhitehallImporter::CreateImageRevision.call(record, image, memo.map(&:filename))
+                end
```

Comment:
There's quite a lot going on in this assignment it looks like it might be working more like a map than a reduce unless I'm missing something.

Reply:
Ahh, I just went in to refactor this and then remembered why it's a `reduce`. It's to keep track of which filenames have been used, so that we avoid having any duplicate filenames (this is passed in the `memo.map(&:filename)` part).

Reply:
Oh yes shoot sorry that rings a bell. 

Maybe we can do something like:

```
    def find_or_create_file_attachment_revisions(file_attachments)
      file_attachments.reduce([]) do |memo, file_attachment|
        already_imported = WhitehallMigration::AssetImport.find_by(original_asset_url: file_attachment["url"])
        
        revision = if already imported
                     already_imported.file_attachment_revision
                   else
                     WhitehallImporter::CreateFileAttachmentRevision
                       .call(record, file_attachment,memo.map(&:filename))
                   end
        memo << revision
      end
    end
```